docs/tutorial/streamer_design.md

```diff
@@ -1,29 +1,398 @@
+
+
+The `temporal_addrgen_unit_params` component with `loop_dim` generates the number of temporal loop bounds. Since it's just 1, then we only have 1 temporal loop bound.
+```hjson
+temporal_addrgen_unit_params: {
+  loop_dim: 1,
+}
+```
+
+The `fifo_reader_params` and `fifo_writer_params` configure the FIFOs. Take note that the FIFOs are directly connected to the accelerator interface. It consists of the `fifo_width` which configures the datawidth of the accelerator interface ports. It also has the FIFO depths `fifo_depth` which configures the depth of the FIFOs.
+
+```
+fifo_reader_params: {
+  fifo_width: [256, 256],
+  fifo_depth: [8, 8],
+}
+
+fifo_writer_params: {
+  fifo_width: [512],
+  fifo_depth: [8],
+}
+```
+
+The `data_reader_params` and `data_writer_params` effectively configures the TCDM interface ports. It consists of `tcdm_ports_num` which control how many TCDM ports are needed. The `spatial_bounds` pertain to the spatial unrolling factors (your parfor) for each data mover. The `spatial_dim` is the dimension of spatial unrolling factors (your parfor) for each data mover. The `element_width` is the single data element width for each data mover, useful for generating spatial addresses.
+
+```
+data_reader_params:{
+  tcdm_ports_num: [4, 4],
+  spatial_bounds: [[4], [4]],
+  spatial_dim: [1,1],
+  element_width: [64,64],
+}
+
+data_writer_params:{
+  tcdm_ports_num: [8],
+  spatial_bounds: [[8]],
+  spatial_dim: [1],
+  element_width: [64],
+}
+```
+
+The number of elements in a list pertain to how many data movers there are. For example, there are 2 elements in the `data_reader_params` and therefore it instantiates two read data movers. There is only 1 element in the `data_writer_params` and hence only instantiates 1 write data mover.
+
+Finally there is the `stationarity` configuration which is for stationarity for each data mover. If the stationarity bit is set, the innermost loop for that data mover is set to 1. This does not instantiate anything but rather affects the loop bound for a data mover. Basically, it only fixes the loop bound to 1. This is a special case scenario only. In the SNAX ALU, this is 0. 
+
+## SNAX ALU Streamer CSRs
+
+Based on the configuration file we expect the CSR registers to have:
+
+- 1 register for a single loop bound as specified in the `temporal_addrgen_unit_params -> loop_dim = 1`. If the number is 2 then we would have 2 loop bounds.
+- 3 temporal stride registers, 3 spatial stride registers, and 3 base addres spointers since we have 3 data movers in total (2 read data movers and 1 write data movers).
+- 1 start register to start the streamer.
+- 1 performance counte register for how many cycles the streamer was active.
+
+Below is a tabulated version of the CSRs of the streamers with the register address, type, and description:
+
+|  register name      |  register addr  |   type  |                   description                        |
```

Comment:
```suggestion
|  register name      |  register offs  |   type  |                   description                        |
```
Is this not an offset on a base address?

Reply:
Good point! Yes~